### ISSUE TITLE
Client exception if changing contents before FW application loaded first time.

### ISSUE THREAD

**Issue by gilberto-torrezan** [vaadin/multiplatform-runtime#13]
@caalador commented on [Fri Jun 15 2018](https://github.com/vaadin/multiplatform-runtime-internal/issues/87)

If you have (for instance) a Flow Tabs that can be used to navigate to a new view containing another embedded FW application and navigate before the application has loaded the first time you get a:

`(TypeError) : $0.firstElementChild.setResponse is not a function`

as a error notification and the console stack trace:

```
SEVERE: Error sending hierarchy change eventscom.google.gwt.event.shared.UmbrellaException: Exception caught: (TypeError) : Cannot set property '__listener' of null
	at Unknown.Hg(AppWidgetset-0.js)
	at Unknown.Lg(AppWidgetset-0.js)
	at Unknown.Rg(AppWidgetset-0.js)
	at Unknown.Sx(AppWidgetset-0.js)
	at Unknown.new Vx(AppWidgetset-0.js)
	at Unknown.qx(AppWidgetset-0.js)
	at Unknown.EFb(AppWidgetset-0.js)
	at Unknown.SFb(AppWidgetset-0.js)
	at Unknown.uRb(AppWidgetset-0.js)
	at Unknown.pRb(AppWidgetset-0.js)
	at Unknown.WQb(AppWidgetset-0.js)
	at Unknown.lRb(AppWidgetset-0.js)
	at Unknown.LBb(AppWidgetset-0.js)
	at Unknown.yCb(AppWidgetset-0.js)
	at Unknown.Hb(AppWidgetset-0.js)
	at Unknown.eval(AppWidgetset-0.js)
	at Unknown.oi(AppWidgetset-0.js)
	at Unknown.ri(AppWidgetset-0.js)
	at Unknown.eval(AppWidgetset-0.js)
Caused by: com.google.gwt.core.client.JavaScriptException: (TypeError) : Cannot set property '__listener' of null
	at Unknown.Lkb(AppWidgetset-0.js)
	at Unknown.anb(AppWidgetset-0.js)
	at Unknown.new yEd(AppWidgetset-0.js)
	at Unknown.qEd(AppWidgetset-0.js)
	at Unknown.vEd(AppWidgetset-0.js)
	at Unknown.XDb(AppWidgetset-0.js)
	at Unknown.zx(AppWidgetset-0.js)
	at Unknown.qx(AppWidgetset-0.js)
	at Unknown.EFb(AppWidgetset-0.js)
	at Unknown.SFb(AppWidgetset-0.js)
	at Unknown.uRb(AppWidgetset-0.js)
	at Unknown.pRb(AppWidgetset-0.js)
	at Unknown.WQb(AppWidgetset-0.js)
	at Unknown.lRb(AppWidgetset-0.js)
	at Unknown.LBb(AppWidgetset-0.js)
	at Unknown.yCb(AppWidgetset-0.js)
	at Unknown.Hb(AppWidgetset-0.js)
	at Unknown.eval(AppWidgetset-0.js)
	at Unknown.oi(AppWidgetset-0.js)
	at Unknown.ri(AppWidgetset-0.js)
	at Unknown.eval(AppWidgetset-0.js)
```

---

@caalador commented on [Fri Jun 15 2018](https://github.com/vaadin/multiplatform-runtime-internal/issues/87#issuecomment-397586825)

For a simple test use FilteringTable or PopupButton demo.

---

@caalador commented on [Tue Jun 19 2018](https://github.com/vaadin/multiplatform-runtime-internal/issues/87#issuecomment-398287473)

The faster load after #92 makes this less of an issue and needs actual fast clicking to happen.

**Comment by pleku** [comment]
It is possible to block the user from interacting with the UI with the loading indicator in Flow, by setting it to eg. dim the whole page for the loading time, and thus block user interaction to the components.

**Comment by eriklumme** [comment]
I get this without interacting with the web page at all

**Comment by pleku** [comment]
My bet is that this is resolved by the fix for #72. In case it is not, please comment here or open a new issue.

**Comment by johannest** [comment]
Seems to reproduce still. Bigger multimodule mpr V7 - V14 project:
V 7.7.17 and 14.4.4, mpr-v7-2.0.6
Happens usually when reloading the page containing a legacy view.

`Uncaught TypeError: $0.firstElementChild.setResponse is not a function
    at Object.eval (eval at cu (client-C4B53966888EAB6CBD0D20BD246409CB.cache.js:1), <anonymous>:3:22)`

Un-obfuscated stack trace
```
client-3E85BB70F7985670DD3EF3C6DD5C2A2E.cache.js:1894 Uncaught TypeError: $0.firstElementChild.setResponse is not a function
    at Object.eval (eval at com_vaadin_client_flow_ExecuteJavaScriptProcessor_$invoke__Lcom_vaadin_client_flow_ExecuteJavaScriptProcessor_2_3Ljava_lang_String_2Lcom_vaadin_client_flow_collection_JsArray_2Lcom_vaadin_client_flow_collection_JsMap_2V (client-3E85BB70F7985670DD3EF3C6DD5C2A2E.cache.js:6359), <anonymous>:3:22)
    at com_vaadin_client_flow_ExecuteJavaScriptProcessor_$invoke__Lcom_vaadin_client_flow_ExecuteJavaScriptProcessor_2_3Ljava_lang_String_2Lcom_vaadin_client_flow_collection_JsArray_2Lcom_vaadin_client_flow_collection_JsMap_2V (client-3E85BB70F7985670DD3EF3C6DD5C2A2E.cache.js:6360)
    at com_vaadin_client_flow_ExecuteJavaScriptProcessor_$handleInvocation__Lcom_vaadin_client_flow_ExecuteJavaScriptProcessor_2Lelemental_json_JsonArray_2V (client-3E85BB70F7985670DD3EF3C6DD5C2A2E.cache.js:6349)
    at com_vaadin_client_flow_ExecuteJavaScriptProcessor_$execute__Lcom_vaadin_client_flow_ExecuteJavaScriptProcessor_2Lelemental_json_JsonArray_2V (client-3E85BB70F7985670DD3EF3C6DD5C2A2E.cache.js:6287)
    at com_vaadin_client_communication_MessageHandler_$lambda$4__Lcom_vaadin_client_communication_MessageHandler_2Lelemental_json_JsonObject_2V (client-3E85BB70F7985670DD3EF3C6DD5C2A2E.cache.js:5320)
    at com_vaadin_client_communication_MessageHandler$lambda$4$Type_MessageHandler$lambda$4$Type__Lcom_vaadin_client_communication_MessageHandler_2Lelemental_json_JsonObject_2V.com_vaadin_client_communication_MessageHandler$lambda$4$Type_flush__V [as flush__V] (client-3E85BB70F7985670DD3EF3C6DD5C2A2E.cache.js:5573)
    at com_vaadin_client_flow_reactive_Reactive_flush__V (client-3E85BB70F7985670DD3EF3C6DD5C2A2E.cache.js:9684)
    at com_vaadin_client_communication_MessageHandler_$processMessage__Lcom_vaadin_client_communication_MessageHandler_2Lcom_vaadin_client_ValueMap_2Ljava_lang_Object_2DV (client-3E85BB70F7985670DD3EF3C6DD5C2A2E.cache.js:5382)
    at com_vaadin_client_communication_MessageHandler$lambda$0$Type_MessageHandler$lambda$0$Type__Lcom_vaadin_client_communication_MessageHandler_2Lcom_vaadin_client_ValueMap_2Ljava_lang_Object_2DV.com_vaadin_client_communication_MessageHandler$lambda$0$Type_execute__V [as execute__V] (client-3E85BB70F7985670DD3EF3C6DD5C2A2E.cache.js:5540)
    at com_vaadin_client_DependencyLoader_runWhenEagerDependenciesLoaded__Lcom_vaadin_client_Command_2V (client-3E85BB70F7985670DD3EF3C6DD5C2A2E.cache.js:2332)
    at com_vaadin_client_communication_MessageHandler_$handleJSON__Lcom_vaadin_client_communication_MessageHandler_2Lcom_vaadin_client_ValueMap_2V (client-3E85BB70F7985670DD3EF3C6DD5C2A2E.cache.js:5250)
    at com_vaadin_client_communication_MessageHandler_$handleMessage__Lcom_vaadin_client_communication_MessageHandler_2Lcom_vaadin_client_ValueMap_2V (client-3E85BB70F7985670DD3EF3C6DD5C2A2E.cache.js:5267)
    at com_vaadin_client_communication_XhrConnection$XhrResponseHandler_XhrConnection$XhrResponseHandler__Lcom_vaadin_client_communication_XhrConnection_2V.com_vaadin_client_communication_XhrConnection$XhrResponseHandler_onSuccess__Lcom_google_gwt_xhr_client_XMLHttpRequest_2V [as onSuccess__Lcom_google_gwt_xhr_client_XMLHttpRequest_2V] (client-3E85BB70F7985670DD3EF3C6DD5C2A2E.cache.js:6208)
    at com_vaadin_client_gwt_elemental_js_util_Xhr$Handler_Xhr$Handler__Lcom_vaadin_client_gwt_elemental_js_util_Xhr$Callback_2V.com_vaadin_client_gwt_elemental_js_util_Xhr$Handler_onReadyStateChange__Lcom_google_gwt_xhr_client_XMLHttpRequest_2V [as onReadyStateChange__Lcom_google_gwt_xhr_client_XMLHttpRequest_2V] (client-3E85BB70F7985670DD3EF3C6DD5C2A2E.cache.js:10006)
    at XMLHttpRequest.<anonymous> (client-3E85BB70F7985670DD3EF3C6DD5C2A2E.cache.js:1583)
    at com_google_gwt_core_client_impl_Impl_apply__Ljava_lang_Object_2Ljava_lang_Object_2Ljava_lang_Object_2Ljava_lang_Object_2 (client-3E85BB70F7985670DD3EF3C6DD5C2A2E.cache.js:916)
    at com_google_gwt_core_client_impl_Impl_entry0__Ljava_lang_Object_2Ljava_lang_Object_2Ljava_lang_Object_2Ljava_lang_Object_2 (client-3E85BB70F7985670DD3EF3C6DD5C2A2E.cache.js:955)
    at XMLHttpRequest.<anonymous> (client-3E85BB70F7985670DD3EF3C6DD5C2A2E.cache.js:943)
```

**Comment by johannest** [comment]
When using a production build this error disappears.
